labs/ipfs-naming/api-cli/names-operations.mdx

```diff
@@ -0,0 +1,351 @@
+
+The [Scaleway Command Line Interface (CLI)](https://github.com/scaleway/scaleway-cli) allows you to create and interact with your names. You will be able to create names, manage them, add/delete value within them.
+
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+    - You have an account and are logged into the [Scaleway console](https://console.scaleway.com/)
+    - You have generated your [API key](/identity-and-access-management/iam/how-to/create-api-keys/)
+    - You have downloaded and installed the [Scaleway CLI v2](https://github.com/scaleway/scaleway-cli#scaleway-cli-v2)
+</Message>
+
+## Configuring the CLI
+
+1. Open a terminal and enter the following command, to configure the Scaleway CLI on your machine:
+
+`scw init`
+
+If you have already configured the CLI, you will see a message like the following:
+
+ ```
+ Current config is located at /path/to/config.yaml
+ access_key: <YOUR ACCESS KEY>
+ secret_key: <YOUR SECRET KEY>
+ default_organization_id: <YOUR ORGANIZATION ID>
+ default_project_id: <YOUR PROJECT ID>
+ default_region: <YOUR DEFAULT REGION, eg fr-par>
+ default_zone: <YOUR DEFAULT ZONE, eg fr-par-1>
+
+ Do you want to override the current config? (Y/n):
+```
+<Message type="note">
+    As the project is available on pl-waw, nl-ams and fr-par, you can set the region to pl-waw, nl-ams or fr-par.
+</Message>
+
+2. Unless you want to change your configuration, you can type `n` and go directly to the [Creating a Name](#creating-a-name) step of this tutorial.
+
+If you have not yet configured the CLI, you will be guided through the configuration process via a series of questions:
+
+ ```
+ Enter a valid secret-key or an email
+    ```
+3. Enter either the email address for your account (you will then be asked for your password), or your [API secret key](/console/project/how-to/create-ssh-key/):
+ ```
+ To improve this tool we rely on diagnostic and usage data.
+ Sending such data is optional and can be disabled at any time by running "scw config set send-telemetry=false".
+ Do you want to send usage statistics and diagnostics? (Y/n):
+    ```
+
+Type `y` or `n` and hit enter.
+
+ ```
+ To fully enjoy Scaleway CLI we recommend you install autocomplete support in your shell.
+ Do you want to install autocomplete? (Y/n):
+    ```
+Type `y` or `n` and hit enter.
+
+ ```
+ To enable autocomplete, scw needs to update your shell configuration.
+ What type of shell are you using (default: zsh):
+    ```
+4. Hit enter if the default shell type is already correct for you. Otherwise, enter your shell type and hit enter.
+ ```
+ To enable autocomplete we need to append to /Users/yourusername/.zshrc the following lines:
+ # Scaleway CLI autocomplete initialization.
+ eval "$(scw autocomplete script shell=zsh)"
+ Do you want to proceed with these changes? (Y/n):
+```
+
+Type `y` or `n` and hit enter. This should complete the configuration process:
+
+```
+ Initialization completed with success.
+```
```

Comment:
```suggestion
    If you have not yet configured the CLI, you will be guided through the configuration process via a series of questions:
    
     ```
     Enter a valid secret-key or an email
        ```
    3. Enter either the email address for your account (you will then be asked for your password), or your [API secret key](/console/project/how-to/create-ssh-key/):
     ```
     To improve this tool we rely on diagnostic and usage data.
     Sending such data is optional and can be disabled at any time by running "scw config set send-telemetry=false".
     Do you want to send usage statistics and diagnostics? (Y/n):
        ```
    
    Type `y` or `n` and hit enter.
    
     ```
     To fully enjoy Scaleway CLI we recommend you install autocomplete support in your shell.
     Do you want to install autocomplete? (Y/n):
        ```
    Type `y` or `n` and hit enter.
    
     ```
     To enable autocomplete, scw needs to update your shell configuration.
     What type of shell are you using (default: zsh):
        ```
    4. Hit enter if the default shell type is already correct for you. Otherwise, enter your shell type and hit enter.
     ```
     To enable autocomplete we need to append to /Users/yourusername/.zshrc the following lines:
     # Scaleway CLI autocomplete initialization.
     eval "$(scw autocomplete script shell=zsh)"
     Do you want to proceed with these changes? (Y/n):
    ```
    
    Type `y` or `n` and hit enter. This should complete the configuration process:
    
    ```
     Initialization completed with success.
    ```
```